scripts/add-changelog.ts

```diff
@@ -0,0 +1,87 @@
+import { readdir, readFile, writeFile } from 'fs/promises';
+import { getPackageVersion } from './get-package-version.js';
+
+async function getChangelogWithVersion(v?: string): Promise<string> {
+  v = v || (await getPackageVersion());
+  const changelog = await readFile('CHANGELOG.md', { encoding: 'utf8' });
+  const [, olderLogs] = changelog.split(`\n# ${v}`);
+  if (!olderLogs) {
+    throw new Error(`Can not find version ${v} in CHANGELOG.md`);
+  }
+  let logs = olderLogs.split('\n# ')[0];
+  logs = '\n' + logs.slice(logs.indexOf('\n##') + 1);
+  logs = logs.replace(/## /g, '### ');
+
+  const date = new Date();
+  const day = date.toLocaleString('default', { day: '2-digit' });
+  const month = date.toLocaleString('default', { month: 'long' });
+  const year = date.getFullYear();
+
+  const headerWithVersion = `\n## ${v} [Core Modules] - ${month} ${day}, ${year}`;
+
+  return [headerWithVersion, logs].join('\n');
+}
+
+async function getReleaseNotesFilePath(): Promise<string> {
+  const majorVersion = (await getPackageVersion()).split('.')[0];
```

Comment:
[req] Package version can be read directly using `import {version} from './package.json';`

Reply:
I get the following error if I directly import a version:
```
Named imports from a JSON file into an ECMAScript module are not allowed when 'module' is set to 'Node16'
```

Reply:
Oh... After searching a bit, I found https://github.com/microsoft/TypeScript/issues/60589, and it looks like it is indeed decided to not be supported with `Node16`.